main/src/cgeo/geocaching/Image.java

```diff
@@ -15,27 +17,92 @@
 
 import java.io.File;
 
+/**
+ * Represent an Image along with Title and Description.
+ * Mostly used for representing image in/from logs.
+ */
 public class Image implements Parcelable {
-    private final String url;
-    private final String title;
-    private final String description;
+    @NonNull private Uri uri;
```

Comment:
>  we should have a static Image.builder() create an mutable object, with setters

is `object` an `ImageBuilder` class in this context?

So it could be ?
```
class ImageBuilder {
    @NonNull private Uri uri;
    @Nullable private String title;
    @Nullable private String description;

    public Image build() {...};
    public void setTitle() {...};
    [then only setters...]
}

class Image {
    @NonNull final private Uri uri;
    @Nullable final private String title;
    @Nullable final private String description;

    public ImageBuilder builder() {...};
    [then current functions...]
}
```
May I put the two classes in the same file?